src/frontend/src/optimizer/plan_node/generic/log_scan.rs

```diff
@@ -0,0 +1,156 @@
+    pub fn output_column_ids(&self) -> Vec<ColumnId> {
+        self.output_col_idx
+            .iter()
+            .map(|i| self.get_table_columns()[*i].column_id)
+            .collect()
+    }
+
+    pub fn primary_key(&self) -> &[ColumnOrder] {
+        &self.table_desc.pk
+    }
+
+    pub(crate) fn column_names_with_table_prefix(&self) -> Vec<String> {
+        let mut out_column_names: Vec<_> = self
+            .output_col_idx
+            .iter()
+            .map(|&i| format!("{}.{}", self.table_name, self.get_table_columns()[i].name))
+            .collect();
+        out_column_names.push(format!("{}.{}", self.table_name, OP_NAME));
+        out_column_names
+    }
+
+    pub(crate) fn column_names(&self) -> Vec<String> {
+        let mut out_column_names: Vec<_> = self
+            .output_col_idx
+            .iter()
+            .map(|&i| self.get_table_columns()[i].name.clone())
+            .collect();
+        out_column_names.push(OP_NAME.to_string());
+        out_column_names
+    }
+
+    pub fn distribution_key(&self) -> Option<Vec<usize>> {
+        let tb_idx_to_op_idx = self
+            .output_col_idx
+            .iter()
+            .enumerate()
+            .map(|(op_idx, tb_idx)| (*tb_idx, op_idx))
+            .collect::<HashMap<_, _>>();
+        self.table_desc
+            .distribution_key
+            .iter()
+            .map(|&tb_idx| tb_idx_to_op_idx.get(&tb_idx).cloned())
+            .collect()
+    }
+
+    /// Create a logical scan node for log table scan
+    pub(crate) fn new(
+        table_name: String,
+        output_col_idx: Vec<usize>,
+        table_desc: Rc<TableDesc>,
+        ctx: OptimizerContextRef,
+        old_epoch: u64,
+        new_epoch: u64,
+    ) -> Self {
+        Self {
+            table_name,
+            output_col_idx,
+            table_desc,
+            chunk_size: None,
+            ctx,
+            old_epoch,
+            new_epoch,
+        }
+    }
+
+    pub(crate) fn columns_pretty<'a>(&self, verbose: bool) -> Pretty<'a> {
+        Pretty::Array(
+            match verbose {
+                true => self.column_names_with_table_prefix(),
+                false => self.column_names(),
+            }
+            .into_iter()
+            .map(Pretty::from)
+            .collect(),
+        )
+    }
+
+    pub(crate) fn schema(&self) -> Schema {
+        let mut fields: Vec<_> = self
+            .output_col_idx
+            .iter()
+            .map(|tb_idx| {
+                let col = &self.get_table_columns()[*tb_idx];
+                Field::from_with_table_name_prefix(col, &self.table_name)
+            })
+            .collect();
+        fields.push(Field::with_name(
+            OP_TYPE,
+            format!("{}.{}", &self.table_name, OP_NAME),
+        ));
+        Schema { fields }
+    }
+
+    pub(crate) fn ctx(&self) -> OptimizerContextRef {
+        self.ctx.clone()
+    }
+
+    pub fn get_table_columns(&self) -> Vec<ColumnDesc> {
```

Comment:
I think this `get_table_columns` method can be removed. In where we use it, we can just use `self.table_desc.columns` directly.